contracts/bridge/x/v4/Registration.sol

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+import {Core} from "./Core.sol";
+
+contract Registration {
```

Comment:
Should this be RegistrationV4? It will save changing the file name when we move from v4 to v5.

Reply:
@drinkcoffee, yeah I also thought of that and was on the fence once v5 is backwards compatible with this version, however thinking better, we might have to support similar functionalities for erc-1155 token (which are not available on v4), meaning, we'll need to deploy another contract. I'll rename it to `Registrationv4`, then. Cheers.